sdks/python/apache_beam/io/fileio.py

```diff
@@ -195,6 +207,8 @@ def open(self, temp_path):
     The returned file handle is passed to ``write_[encoded_]record`` and
     ``close``.
     """
+    if self._file_system is None:
+      self._file_system = get_filesystem(temp_path)
```

Comment:
Should not this be `get_filesystem(self.file_path_prefix.get())`? Otherwise the type of `self._file_system` might be different depending on which function is called first and its parameters.